packages/components/cascader/src/cascader.vue

```diff
@@ -0,0 +1,658 @@
+            @close="deleteTag(tag)"
+          >
+            <template v-if="tag.isCollapseTag === false">
+              <span>{{ tag.text }}</span>
+            </template>
+            <template v-else>
+              <el-tooltip
+                :disabled="popperVisible || !collapseTagsTooltip"
+                :fallback-placements="['bottom', 'top', 'right', 'left']"
+                placement="bottom"
+                effect="light"
+              >
+                <template #default>
+                  <span>{{ tag.text }}</span>
+                </template>
+                <template #content>
+                  <div :class="nsCascader.e('collapse-tags')">
+                    <div
+                      v-for="(tag2, idx) in allPresentTags.slice(1)"
+                      :key="idx"
+                      :class="nsCascader.e('collapse-tag')"
+                    >
+                      <el-tag
+                        :key="tag2.key"
+                        class="in-tooltip"
+                        :type="tagType"
+                        :size="tagSize"
+                        :hit="tag2.hitState"
+                        :closable="tag2.closable"
+                        disable-transitions
+                        @close="deleteTag(tag2)"
+                      >
+                        <span>{{ tag2.text }}</span>
+                      </el-tag>
+                    </div>
+                  </div>
+                </template>
+              </el-tooltip>
+            </template>
+          </el-tag>
+          <input
+            v-if="filterable && !isDisabled"
+            v-model="searchInputValue"
+            type="text"
+            :class="nsCascader.e('search-input')"
+            :placeholder="presentText ? '' : inputPlaceholder"
+            @input="(e) => handleInput(searchInputValue, e as KeyboardEvent)"
+            @click.stop="togglePopperVisible(true)"
+            @keydown.delete="handleDelete"
+            @compositionstart="handleComposition"
+            @compositionupdate="handleComposition"
+            @compositionend="handleComposition"
+          />
+        </div>
+      </div>
+    </template>
+
+    <template #content>
+      <el-cascader-panel
+        v-show="!filtering"
+        ref="panel"
+        v-model="checkedValue"
+        :options="options"
+        :props="props.props"
+        :border="false"
+        :render-label="$slots.default"
+        @expand-change="handleExpandChange"
+        @close="$nextTick(() => togglePopperVisible(false))"
+      />
+      <el-scrollbar
+        v-if="filterable"
+        v-show="filtering"
+        ref="suggestionPanel"
+        tag="ul"
+        :class="nsCascader.e('suggestion-panel')"
+        :view-class="nsCascader.e('suggestion-list')"
+        @keydown="handleSuggestionKeyDown"
+      >
+        <template v-if="suggestions.length">
+          <li
+            v-for="item in suggestions"
+            :key="item.uid"
+            :class="[
```

Comment:
## Suggest

Same here

Reply:
Because it is difficult to convert the loop variable into a calculated attribute, skip it first.

Reply:
You're right.